lib/osf-components/addon/components/file-renderer/component.ts

```diff
@@ -52,15 +51,23 @@ export default class FileRenderer extends Component {
     @computed('download', 'params', 'version')
     get downloadUrl(): string {
         const { download, params, version } = this;
-
-        return `${download}?${$.param({ ...params, ...(version ? { version } : {}) })}`;
+        if (!download) {
+            return '';
+        }
+        const downloadLink = new URL(download);
+        downloadLink.searchParams.set('direct', params.direct || '');
+        downloadLink.searchParams.set('mode', params.mode);
+        downloadLink.searchParams.set('version', version?.toString() || '');
```

Comment:
Honestly, I'm not even sure why there's a version parameter here at all, but the people to ask no longer work here. It would prooooobably be safe to remove, but if we wanted to keep it, making it optional seems the way to go.

Reply:
I take it back; the version parameter is almost certainly for viewing other revisions of the file and would totally be necessary. So keep that, but only use it if there's one to use.